### PR TITLE
Export Volume Annotations

### LINKED ISSUE #5203: Tiff Export for Volume Annotations
https://github.com/scalableminds/webknossos/pull/5195 creates the possibility to export data as tiff stacks.
It would be helpful for annotation pipelines if this could include volume annotations created by the webKnossos user.

The long-running job would have to get the volume data from the tracingstore (or fossildb directly?), unpack it, and mix it with the fallback segmentation layer data.

Should probably include the possibility to select the version of the volume tracing.

### PR DESCRIPTION
**Pull request by fm3**
image

Note that #5252 should be merged first (some of the bugs fixed there could crop up here too during testing)

### TODO (includes changes in https://github.com/scalableminds/webknossos-worker/pull/6/ )
 - [x] fetch volume annotations
 - [x] merge with fallback layer
 - [x] pass result to cuber
 - [x] are compound tracings cached long genough? → increased to 70 min, created follow-up issue #5255 
 - [x] use glob instead of exists
 - [x] pass version
 - [x] pass token only for this job
 - [x] pass tracing id from frontend
 - [x] don’t include fallback layer if it is unlinked
 - [x] test resulting files with imagemagick (`convert 000002.tiff -evaluate multiply 200 -depth 8 8.tif`)
 - [x] adapt job description in list view
 - [x] disable buttons already in frontend if mag1 is not present
 - [x] ensure saved state
 - [x] clean up intermediate files? (let’s leave them in for the moment, for debugging. clean-up job issue #5202)

### Steps to test:
- set up https://github.com/scalableminds/webknossos-worker/pull/6/
- set `jobsEnabled = true` in application.conf
- open volume annotation (also try hybrid, skeleton, view mode)
- brush some, create bbox
- click export tif button, modal should show annotation layer
- zip download from jobs list should include segmentation tifs with the annotations
- modals should not contain typos

### Issues:
- fixes #5203 

------
- [x] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [x] Ready for review

### CODE REVIEW

**Comment by fm3** [comment]
Another idea: might be worth to save the “referrer” link with the job (e.g. to go back to the source (compound) annotation the tiffs were exported from. The tracing id is not sufficient for this lookup in compound case.

File: frontend/javascripts/oxalis/view/settings/export_bounding_box_modal.js [hunk 1]
```diff
@@ -2,54 +2,82 @@
...
   };
 
-  const layerNames = dataset.dataSource.dataLayers.map(layer => {
-    const nameIfFromDataset = layer.category === "color" || !hasVolumeTracing ? layer.name : null;
-    const nameIfVolume =
-      hasVolumeTracing && layer.category === "segmentation" && layer.fallbackLayerInfo != null
-        ? layer.fallbackLayerInfo.name
-        : null;
-    return nameIfFromDataset || nameIfVolume;
+  const hasMag1 = (layer: APIDataLayer) => layer.resolutions.map(r => Math.max(...r)).includes(1);
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/settings/export_bounding_box_modal.js:42]: you can use the ResolutionInfo abstraction like this:

```suggestion
  const hasMag1 = (layer: APIDataLayer) => getResolutionInfo(layer.resolutions).hasIndex(0);
```

`getResolutionInfo` needs to be imported from `oxalis/model/accessors/dataset_accessor`

File: frontend/javascripts/oxalis/view/settings/export_bounding_box_modal.js [hunk 2]
```diff
@@ -2,54 +2,82 @@
...
-          onClick={() => handleStartExport(layerName)}
-          disabled={startedExports.includes(layerName)}
+          key={exportKey(layerInfos)}
+          onClick={() => handleStartExport(layerInfos)}
+          disabled={startedExports.includes(exportKey(layerInfos)) || !layerInfos.hasMag1}
         >
-          {layerName}
-          {startedExports.includes(layerName) ? " (started)" : null}
+          {layerInfos.displayName}
+          {!layerInfos.hasMag1 ? " (resolution 1 missing)" : ""}
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/settings/export_bounding_box_modal.js:79]: can you create a follow-up issue to support exports where mag1 is missing? is it difficult to support this, too?

File: frontend/javascripts/oxalis/view/settings/export_bounding_box_modal.js [hunk 3]
```diff
@@ -2,54 +2,82 @@
...
+      return {
+        displayName: layer.name,
+        layerName: layer.name,
+        tracingId: null,
+        tracingVersion: null,
+        hasMag1: hasMag1(layer),
+      };
+    if (layer.fallbackLayerInfo != null)
+      return {
+        displayName: "Volume annotation with fallback segmentation",
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/settings/export_bounding_box_modal.js:55]: nice :+1:

**Review by philippotto — COMMENTED** [review]
front-end looks great! will test now.

File: frontend/javascripts/oxalis/view/settings/export_bounding_box_modal.js (hunk 2 above)
Comment (fm3) [reply frontend/javascripts/oxalis/view/settings/export_bounding_box_modal.js:79]: https://github.com/scalableminds/webknossos/issues/5260
Shouldn’t be too difficult, but didn’t strike me as highest-priority at this time

File: frontend/javascripts/oxalis/view/settings/export_bounding_box_modal.js (hunk 1 above)
Comment (fm3) [reply frontend/javascripts/oxalis/view/settings/export_bounding_box_modal.js:42]: Done :) Though it reminds me that we should really tackle #4762 and #4893 (probably we should do this together as it is so interlinked in front-end and back-end)

**Review by youri-k — APPROVED** [review]
I have not tested it, but the backend code looks good :tada:

**Pull request merged**